kotest-framework/kotest-framework-engine/src/commonMain/kotlin/io/kotest/engine/test/TestCaseExecutor.kt

```
@@ -86,7 +86,7 @@ internal class TestCaseExecutor(
             timeMark,
             listOfNotNull(
                InvocationTimeoutInterceptor,
-               if (platform == Platform.JVM && testCase.config.testCoroutineDispatcher) TestDispatcherInterceptor() else null,
+               if (platform == Platform.JVM && testCase.config.testCoroutineDispatcher) createTestDispatcherInterceptor() else null,
                if (useCoroutineTestScope) TestCoroutineInterceptor() else null,
```

... this one. But in that case, we'd need to remove `useCoroutineTestScope`.

Of course, we could throw an exception if a `coroutineTestScope` was requested but the platform does not support it. But I wonder if it would be easier to have the entire logic here in one place instead of distributing it across an expect/actual hierarchy. It seems to be a temporary state waiting for a fix, after all.

---

Yes it'll be removed in that other PR.

---

Don't the two PRs now contradict each other? Doesn't the code added here collide with the removal of #4251?

---

I did this first to make it compile. Kotlin 2 didn't like the class not being abstract. This was to get it to compile. The other one I'll rebase after. It's only because I did this first